### DISCUSSION TITLE
Haskell and Ormolu

### DISCUSSION THREAD

**Discussion by George-Spanos** [LazyVim/LazyVim#3325] [Q&A] [answered]
Hello!

I'm recently exploring migrating to LazyVim. I'm trying to set up a haskell dev environment and it seems to me that the code formatter ormolu does not work out of the box. I've search inside comform.nvim issues and there's a recent update there that, from my understanding, should be already present in my config. I've configed haskell as an extra, with nothing custom. And there's no formatting or error feedback from LayVim. 

I also don't see any formatted as a dependency in the Haskell extra for LazyVim. Is the haskell extra expected to format out of the box or is there any config that I need to do? 

Here's a [link](https://github.com/stevearc/conform.nvim/issues/377) to a possibly related issue to this.

Thanks

**Comment by dpetka2001** [comment, marked as answer]
You would have to install `ormolu` via Mason and then configure `conform.nvim` to use it for formatting Haskell files. Something like
```lua
-- ~/.config/nvim/lua/plugins/conform.lua
return {
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        haskell = { "ormolu" },
      },
    },
  },
}
```
Otherwise, `conform` should fallback to LSP formatting provided by `hls` LSP server that you have installed presumably by `ghcup`. I don't do Haskell, so I'm just guessing here based on how LazyVim operates.

**Reply by George-Spanos** [reply]
It was really that simple, indeed. I got confused by reading that it was expected to be a default.

Thank you for taking the time.
